airflow/www/static/js/dag/details/taskInstance/index.tsx

```diff
@@ -54,18 +52,41 @@ const dagId = getMetaValue('dag_id')!;
 interface Props {
   taskId: Task['id'];
   runId: DagRun['runId'];
+  mapIndex: TaskInstanceType['mapIndex'];
+  onSelect: (selectionProps: SelectionProps) => void;
 }
 
-const TaskInstance = ({ taskId, runId }: Props) => {
-  const [selectedRows, setSelectedRows] = useState<number[]>([]);
+const TaskInstance = ({
+  taskId, runId, mapIndex, onSelect,
+}: Props) => {
+  const isMapIndexDefined = !(mapIndex === undefined);
+  const selectedRows: number[] = isMapIndexDefined ? [mapIndex] : [];
   const { data: { dagRuns, groups } } = useGridData();
 
   const storageTabIndex = parseInt(localStorage.getItem(detailsPanelActiveTabIndex) || '0', 10);
   const [preferedTabIndex, setPreferedTabIndex] = useState(storageTabIndex);
 
+  const [instance, setInstance] = useState<TaskInstanceType>();
+
   const group = getTask({ taskId, task: groups });
   const run = dagRuns.find((r) => r.runId === runId);
 
+  useEffect(() => {
+    if (!isMapIndexDefined) {
+      setInstance(group?.instances.find((ti) => ti.runId === runId));
```

Comment:
I don't think we should be using `useEffect` here.  The Details component should probably get the mapped instance info from the API: `/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/{map_index}`

Reply:
By using a new API endpoint for a single mapped instance, we can remove a lot of this `useEffect` logic, add autorefresh, and make sure that navigating to this page with `map_index` in the URl, data will load.